### PR TITLE
Add Configurable Option to Exclude Trigger Characters for Typing Assists

### PR DESCRIPTION
**Pull request by tareknaser**
This PR addresses #16084 by adding a new configuration option: `rust-analyzer.typing.excludeChars`.

cc: @darichey 
### Changes
- Added `rust-analyzer.typing.excludeChars` configuration.
    - Defaults to `None`, meaning all trigger characters (`.=<>{(`) are active by default.
    - Users can specify a string of characters to exclude from triggering assists.
- Updated documentation and configuration schema to reflect this new option.

### CODE REVIEW

**Comment by darichey** [comment]
I think this is a good workaround for people who want some functionality of format-on-type but find the limitations/bugs (like the one presented in the linked issue) annoying. 

I would reword "typing assists" to "format-on-type"

**Comment by tareknaser** [comment]
> I would reword "typing assists" to "format-on-type"

Thanks for the review.

I initially thought `rust-analyzer.typing.excludeChars` made sense because it's similar to `rust-analyzer.typing.autoClosingAngleBrackets.enable`.

Do you think it should be renamed to `rust-analyzer.formatOnType.excludeChars` instead? If so, should we also rename the other configurable option to `rust-analyzer.formatOnType.autoClosingAngleBrackets.enable`?

**Comment by Veykril** [comment]
This setting should replace `rust-analyzer.typing.autoClosingAngleBrackets.enable` imo (the main reason that setting exists is because the typing for angle brackets is bugged atm)

**Comment by tareknaser** [comment]
> This setting should replace `rust-analyzer.typing.autoClosingAngleBrackets.enable` imo

Sure. I’ve just pushed a commit to remove it from the code. I’m not sure if there’s a specific process for deprecation like this. If there is, please let me know and I can update the code

**Review by Veykril — APPROVED** [review]
